app/src/main/java/ch/epfl/qedit/view/quiz/QuizActivity.java

```diff
@@ -8,28 +9,42 @@
 import android.view.Menu;
 import android.view.MenuItem;
 import android.view.View;
-import android.widget.Toast;
 import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.lifecycle.MutableLiveData;
 import androidx.lifecycle.ViewModelProvider;
 import ch.epfl.qedit.R;
 import ch.epfl.qedit.model.Question;
 import ch.epfl.qedit.model.Quiz;
+import ch.epfl.qedit.model.User;
 import ch.epfl.qedit.model.answer.AnswerModel;
+import ch.epfl.qedit.model.answer.MatrixFormat;
+import ch.epfl.qedit.model.answer.MatrixModel;
 import ch.epfl.qedit.util.LocaleHelper;
+import ch.epfl.qedit.view.home.HomeActivity;
 import ch.epfl.qedit.view.util.ConfirmDialog;
 import ch.epfl.qedit.viewmodel.QuizViewModel;
 import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Objects;
 
 public class QuizActivity extends AppCompatActivity implements ConfirmDialog.ConfirmationListener {
+    public static final String PARTICIPANT_ANSWERS = "ch.epfl.qedit.quiz.PARTICIPANT_ANSWERS";
+    public static final String QUESTIONS = "ch.epfl.qedit.quiz.QUESTIONS";
+    public static final String GOOD_ANSWERS = "ch.epfl.qedit.quiz.GOOD_ANSWERS";
+    public static final String CORRECTION = "ch.epfl.qedit.quiz.CORRECTION";
     private QuizViewModel model;
     private Boolean overviewActive;
 
+    private User user;
     private ConfirmDialog validateDialog;
     private Quiz quiz;
+    private Boolean correction;
+    private ArrayList<Integer> correctedQuestions;
```

Comment:
Why do we need to have two array lists ? Looking at the code, it doesn't seem that we ever need to use them both at the same time. And why can we not make them arrays of booleans ?

Reply:
Yes we could use only one of them.
In a first version I needed them to be integer arraylists but now I don't see any problem in going for boolean arraylist

Reply:
Ok, I see, nice ;)